dev/benchmarks/test_apps/stocks/lib/routing/router.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import '../stock_home.dart';
+import '../stock_settings.dart';
+import '../stock_symbol_viewer.dart';
+import 'router_state.dart';
+
+const String _kSettingsPageLocation = '/settings';
+const String _kStockPageLocation = '/stock';
+const String _kHomePageLocation = '/';
+
+class _RouterConfiguration {
+  _RouterConfiguration(this.path, this.browserState);
+  StockRoutePath path;
+  Map<String, dynamic> browserState;
```

Comment:
If this information is part of StockHomePath, it will be problematic when user currently on StockSymbolPath and go back to StockHomePath using Navigator.pop. In the _handlePopPage, we don't know what is the current tab for StockHomePath.

This makes me think, anything that does not reflect to the URL (i.e. browser state) should be global to all the routes.

Reply:
I agree with @goderbauer that this is suboptimal. I don't think we should be encouraging people to use open maps in this way. The whole point of the router API is to make the data structured.

Reply:
But how do we solve the issue when user uses navigator.pop to go back from stocksymbolpath to stockhomepath?
If the tab index is only store of StockHomePath, in the _handlePopPage I only have the information about stocksymbolpath and i do not know how to set the tab index in StockHomePath.

We can probably make the StockHomePath to take optional tab index. If it is not provided, it will use whatever in its current state

Reply:
making the browser state global is a a compromise between imperative API vs browser routing. If we are sure user do not mix one or another, we can make it a state inside StockHomePath. Or we can simply remove this functionality and let developer figure out what is the best way to store browser state. What do you think?